webapp/utils/post_utils.jsx

```diff
@@ -30,3 +30,32 @@ export function getProfilePicSrcForPost(post, timestamp) {
 
     return src;
 }
+
+export function messageHistoryHandler(e) {
+    if ((e.ctrlKey || e.metaKey) && !e.altKey && !e.shiftKey && (e.keyCode === Constants.KeyCodes.UP || e.keyCode === Constants.KeyCodes.DOWN)) {
+        if (this.state.messageText !== '' && this.currentLastMsgIndex === PostStore.getHistoryLength()) {
```

Comment:
He binds `this` to the components he uses this function in. It's a portable member function :P  Interesting but certainly confusing.